memory/memkind_kmem_allocator_test.cc

```diff
@@ -0,0 +1,41 @@
+/*
+* Copyright (c) 2019 Intel Corporation
+*
+* SPDX-License-Identifier: Apache-2.0
+*
+*/
+
+#include <cstdio>
+
+#ifdef MEMKIND
+#include "memkind_kmem_allocator.h"
+#include "test_util/testharness.h"
+
+namespace rocksdb {
+TEST(MemkindKmemAllocatorTest, Allocate) {
+  MemkindKmemAllocator allocator;
+  void* p;
+  try {
+    p = allocator.Allocate(1024);
+  } catch (const std::bad_alloc& e) {
+	return;
+  }
+  ASSERT_NE(p, nullptr);
+  size_t size = allocator.UsableSize(p, 1024);
+  ASSERT_GE(size, 1024);
+  allocator.Deallocate(p);
+}
```

Comment:
It will be interesting to add a test of full DB with block cache to allocate from this allocator.

Reply:
I added a full DB test using the new allocator. The database is populated and a flush is triggered, so the block cache is used for subsequent reads.